src/components/pager/__internal__/pager-navigation-link.component.tsx

```diff
@@ -61,30 +88,22 @@ const PagerNavigationLink = ({
     <StyledPagerLinkStyles
       data-element={`pager-link-${type}`}
       disabled={disabled()}
-      onClick={handleOnCLick}
+      onClick={
+        // Type assertion due to the fact that StyledPagerLinkStyles
+        // will always return a button element
+        handleOnCLick as (
+          ev:
```

Comment:
It might be worth exporting this type from Link and re-using it here

```
export type LinkClickType = (
  ev:
    | React.MouseEvent<HTMLAnchorElement>
    | React.MouseEvent<HTMLButtonElement>
    | React.KeyboardEvent<HTMLAnchorElement>
    | React.KeyboardEvent<HTMLButtonElement>
) => void;
```

Reply:
I agree @robinzigmond, I've actually tinkered a little bit with Link interface to make as you said `anchor props` or `button props` but I had no luck. I didn't put much effort to it though and I'm pretty sure it's possible to be done.

Reply:
I also think that it's enough to export just the `LinkProps` since you can access the `onClick` property type - `LinkProps["onClick"]`.